internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -745,7 +765,12 @@ func (r *BMACReconciler) newSpokeBMH(bmh *bmh_v1alpha1.BareMetalHost, machine *m
 			Namespace: machine.Namespace,
 		}
 		// copy annotations. hardwaredetails annotations is needed for automatic csr approval
-		bmhSpoke.ObjectMeta.Annotations = bmh.ObjectMeta.Annotations
+		// We don't copy all annotations because there are some annotations that should not be
+		// carried over from the hub BMH. The detached annotation is an example.
+		if bmhSpoke.ObjectMeta.Annotations == nil {
```

Comment:
same question? 
host.ObjectMeta.SetAnnotations(annotations)?

Reply:
Same here. The `annotations` parameter is type `map[string]string`. In our case, we just want to set a single annotation and not the entire map.